.travis.yml

```diff
@@ -29,12 +29,15 @@ install:
   - conda info -a
 
   # install dependencies
-  - conda install numpy scipy astropy six nose pip pyyaml line_profiler pycodestyle coveralls h5py psutil
-  - conda install -c conda-forge mpi4py
-  - pip install git+https://github.com/RadioAstronomySoftwareGroup/pyuvdata.git
-  - python setup.py install
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION coveralls
+  - source activate test-environment
+  - conda env update -n test-environment -f environment.yml
+  - pip install .
   - python -c "from pyuvdata import version; version.main()"
   - conda list
+  # removed this, as it's installed via conda. However, it may be that we *want* it to
+  # be from github...
+  #- pip install git+https://github.com/RadioAstronomySoftwareGroup/pyuvdata.git
```

Comment:
I would bet using the conda version is better here. But should we have an integration test on pyuvdata to test against pyuvsim like we do for hera_qm and hera_cal?

Reply:
I think the reason we haven't used the conda version is because sometimes there's very recent development on pyuvdata that pyuvsim needs. But I'm ok with that providing more push to get out small version updates from pyuvdata more frequently (or on an as-needed basis).

We could add testing against pyuvsim on pyuvdata if needed.

Reply:
@aelanman what's your feeling about installing pyuvdata from conda vs the repo for pyuvsim testing?

Reply:
In the past, there have been times we needed to build pyuvsim against a pending branch of pyuvdata, or use a bleeding-edge version that isn't available on conda yet. For instance, I'm planning to add a test in the near future that uses the `ignore_requirements` option on UVBase, which I don't think has been published yet to anaconda, and I'd also like to update pyuvsim to work with the new interp syntax on UVBeam. I don't know how often pyuvdata gets updated on conda.

I think for PRs, we should be testing against the latest merged version of pyuvdata (ie., the repo), and for _releases_ we should be testing against what's been published to conda, since the release is what users will probably install and run.

Reply:
Is there a way to specify which version of pyuvdata to test against per-test on travis? I agree with @aelanman I think -- releases should definitely be tested against released versions. It makes sense that PRs may want to sometimes test against dev versions of pyuvdata.

Although, I think a good case could be made that any PR which is to be pulled into master for `pyuvsim` should rely only on released versions of `pyuvdata`. The developer can test it against dev versions on their own system, but before merging to master, it should probably be built against a released version. This also forces the version requirements to be updated sufficiently within the PR itself (eg. if the new `pyuvsim` PR depends on the newest features of `pyuvdata`, then the PR should be blocked until a new release of `pyuvdata` is made, at which point the `pyuvsim` PR should be updated to have `install_requires = {..., "pyuvdata">=new.release.version"}`).

Reply:
What did we get to on this issue?

Reply:
As discussed on `pyuvsim` telecon, we're going to go ahead and ensure that `pyuvsim` works with _released_ `pyuvdata` versions, and so this will block PRs from merging until the `pyuvdata` release has the necessary feature/behaviour. 

The key to this working is that `pyuvdata` must be released on a reasonably high cadence, (or we end up with the problem we're currently having with `aipy`). @bhazelton reminded us that the best way to do this is to create version-bump PRs on `pyuvdata`.